main.go

```diff
@@ -1,39 +1,69 @@
 package main
 
 import (
+	"bufio"
+	"bytes"
 	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"sort"
+	"strings"
+	"regexp"
+	"time"
+	"strconv"
+
+
+	"github.com/Masterminds/semver/v3"
 	"github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/config"
 	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
 	"github.com/kurtosis-tech/stacktrace"
 	"github.com/sirupsen/logrus"
-	"io/ioutil"
-	"os"
-	"path"
-	"strconv"
-	"time"
 )
 
 const (
 	gitDirname = ".git"
+	gitUsername = "git"
+	originRemoteName = "origin"
+	masterBranchName = "master"
+	emptyPassword = ""
+
+	preReleaseScriptsFilename = ".pre-release-scripts.txt"
+	relScriptsDirpath = "scripts"
+
+	tagsPrefix = "refs/tags/"
+	tagRefSpec = "refs/tags/*:refs/tags/*"
+	headRef = "refs/heads/"
 
 	// The name of the file inside the Git directory which will store when we last fetched (in Unix seconds)
 	lastFetchedFilename = "last-fetch.txt"
-
 	lastFetchedTimestampUintParseBase = 10
 	lastFetchedTimestampUintParseBits = 64
-
 	// How long we'll allow the user to go between fetches to ensure the repo is updated when they're releasing
-	fetchGracePeriod = 30 * time.Second
-
+	fetchGracePeriod = 1 * time.Minute
 	extraNanosecondsToAddToLastFetchedTimestamp = 0
-
 	lastFetchedFileMode = 0644
 
-	developBranchName = "develop"
-	masterBranchName  = "master"
+	relChangelogFilepath = "docs/changelog.md"
+	changelogFileMode = 0644
 
-	originRemoteName = "origin"
+	// Taken from guess-release-version.sh
+	expectedNumTBDHeaderLines = 1
+	tbdHeaderStr = "# TBD"
+	noPreviousVersion = "0.0.0" 
+	semverRegexStr = "^[0-9]+.[0-9]+.[0-9]+$"
+	tbdHeaderRegexStr = "^#\\s*TBD\\s*$"
```

Comment:
I'd extract the `#` in both this line and the one below to be something like `sectionHeaderPrefix`, for dynamicism